include/SFML/System/Vector2.hpp

```diff
@@ -68,14 +73,132 @@ class Vector2
     ////////////////////////////////////////////////////////////
     template <typename U>
     constexpr explicit Vector2(const Vector2<U>& vector);
+    
+    ////////////////////////////////////////////////////////////
+    /// \brief Length of the vector <i><b>(floating-point)</b></i>.
+    ///
+    /// If you are not interested in the actual length, but only in comparisons, consider using lengthSq().
+    ///
+    ////////////////////////////////////////////////////////////
+    T length() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Square of vector's length <i><b>(floating-point)</b></i>.
+    /// 
+    /// Suitable for comparisons, more efficient than length().
+    ///
+    ////////////////////////////////////////////////////////////
+    T lengthSq() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Vector with same direction but length 1 <i><b>(floating-point)</b></i>.
+    /// 
+    /// \pre \c vector is no zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 normalized() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Signed angle from \c *this to \c rhs <i><b>(floating-point)</b></i>.
+    /// 
+    /// \return Angle in degrees in the interval [-180,180].
+    /// The angle determines how much you have to rotate \c *this
+    ///  until it points to the same direction as \c rhs.
+    /// \pre Neither \c *this nor \c rhs is a zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle signedAngleTo(const Vector2& rhs) const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Angle from +X or (1,0) vector <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds to 0 degrees, (0,1) corresponds to 90 degrees.
+    /// 
+    /// \return Angle in degrees in the interval [-180,180].
+    /// \pre \c vector is no zero vector.
+    ///
+    ////////////////////////////////////////////////////////////
+    Angle polarAngle() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Rotate by given angle <i><b>(floating-point)</b></i>.
+    /// 
+    /// The vector (1,0) corresponds 0 degrees, (0,1) corresponds 90 degrees.
+    ///
+    ////////////////////////////////////////////////////////////
+    Vector2 rotatedBy(Angle angle) const;
```

Comment:
```suggestion
    [[nodiscard]] Vector2 rotatedBy(Angle angle) const;
```

Ditto here. I don't feel strongly about it, but wouldn't hurt...